File: Countries/js/5.js
// What countries use only one vowel in their name (the vowel can be used multiple times) For
// example, if the word “BEEKEEPER” were a country, it would be an answer, because it only uses “E”.

let main = () => {

    const fs = require("fs");
    let buffer_arry = fs.readFileSync("Countries/countries.txt", "utf-8");
    let in_file = buffer_arry.split("\n");
    let vowels = ["A", "E", "I", "O", "U"];
    let vowel_arry = [];
    let single_vowel = [];

    let find_vowels = (word) => {

        vowel_arry = [];

        for(let i = 0; i < word.length; i++) {

            for(let j = 0; j < vowels.length; j++) {

                if(word[i] == vowels[j]) {
                    vowel_arry.push(vowels[j]);
                };
            };
        };
    };

    // Function determines if words uses on 1 vowel
    let determine_one = () => {

        // create Set out of vowell_arry. This will remove all duplicates
        single_vowel = [...new Set(vowel_arry)];
        if(single_vowel.length > 1) {
            return false;
        };
        return true;
    };

    // Main function calls
    for(let i = 0; i < in_file.length; i++) {
        find_vowels(in_file[i].toUpperCase());
        if(determine_one()) {
            console.log(in_file[i]);
        };
    };

};

main();

File: Countries/python/5.py
# What countries use only one vowel in their name (the vowel can be used multiple times)
# - For example, if the word “BEEKEEPER” were a country, it would be an answer, 
# because it only uses “E”.
def main():

    in_file = open("../countries.txt", "r")
    for words in in_file:
        one_vowel = {"a": 0, "e": 0, "i": 0, "o": 0, "u": 0}
        for chars in words:
            for keys in one_vowel:
                if(chars.lower() == keys):
                    one_vowel[keys] += 1
        one_vowel_cnt = 0

        # check vowel count for each item in dict
        for value in one_vowel.values():
            if value > 0:
                one_vowel_cnt += 1
        if(one_vowel_cnt == 1):
            print(words, end="")

if(__name__  == "__main__"):
    main()